etna/packages/magby/magby/Magby.py

```diff
@@ -0,0 +1,199 @@
+from typing import List
+import pandas as pd
+from functools import partial
+
+from magby.Magma import *
+
+_session = Session()
+
+class Magby(object):
+    def __init__(self, url: str, token: str):
+        '''
+        Client-facing class to interface with Magma (or Janus - for projects)
+        :param url: str. Magma url
+        :param token: str. Private token
+        '''
+        self._url = url.strip('/')
+        self._token = token
+
+    @property
+    def url(self) -> str:
+        return self._url
+
+    @url.setter
+    def url(self, newUrl: str) -> None:
+        self._url = newUrl
+
+    @property
+    def token(self) -> str:
+        return self._token
+
+    @token.setter
+    def token(self, newToken: str) -> None:
+        self._token = newToken
+
+
+    def retrieve(self,
+                 projectName: str,
+                 modelName: Union[List, str],
+                 recordNames: Union[List, str]='all',
+                 attributeNames: Union[List, str]='all',
+                 dataType: str='meta',
+                 filter='',
+                 **kwargs) -> Union[pd.DataFrame, Dict]:
+        '''
+        Method to retrieve data from Magma
+        :param projectName: str. Project name in Magma
+        :param modelName: List[str] or str. Name(s) of model(s) in Magma
```

Comment:
Curious what use cases you are imagining that involve multiple models. I don't think I have any in magmaR except for the `retrieveMetadata()` endpoint... Is this just in prep for a potential future endpoint like that?

Reply:
yeh, why not